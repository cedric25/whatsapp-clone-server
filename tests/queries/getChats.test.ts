import { createTestClient } from 'apollo-server-testing';
import { ApolloServer, gql } from 'apollo-server-express';
import schema from '../../schema';

// jest.spyOn(Date, 'now')
//   .mockImplementation(() => new Date('2020-03-26T10:00:00.000Z').getTime())

describe('Query.chats', () => {

  it('should fetch all chats', async () => {
    const server = new ApolloServer({ schema });

    console.log('Date.now()', Date.now())

    const { query } = createTestClient(server);

    const res = await query({
      query: gql`
        query GetChats {
          chats {
            id
            name
            picture
            lastMessage {
              id
              content
              createdAt
            }
          }
        }
      `,
    });

    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });
});
