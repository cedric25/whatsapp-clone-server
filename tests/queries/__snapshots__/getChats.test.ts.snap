// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query.chats should fetch all chats 1`] = `
Object {
  "chats": Array [
    Object {
      "id": "1",
      "lastMessage": Object {
        "content": "You on your way?",
        "createdAt": "2020-03-26T06:23:32.724Z",
        "id": "1",
      },
      "name": "Ethan Gonzalez",
      "picture": "https://randomuser.me/api/portraits/thumb/men/1.jpg",
    },
    Object {
      "id": "2",
      "lastMessage": Object {
        "content": "Hey, it's me",
        "createdAt": "2020-03-25T13:43:32.724Z",
        "id": "2",
      },
      "name": "Bryan Wallace",
      "picture": "https://randomuser.me/api/portraits/thumb/men/2.jpg",
    },
    Object {
      "id": "3",
      "lastMessage": Object {
        "content": "I should buy a boat",
        "createdAt": "2020-03-10T07:03:32.724Z",
        "id": "3",
      },
      "name": "Avery Stewart",
      "picture": "https://randomuser.me/api/portraits/thumb/women/1.jpg",
    },
    Object {
      "id": "4",
      "lastMessage": Object {
        "content": "This is wicked good ice cream.",
        "createdAt": "2019-08-06T15:03:32.724Z",
        "id": "4",
      },
      "name": "Katie Peterson",
      "picture": "https://randomuser.me/api/portraits/thumb/women/2.jpg",
    },
  ],
}
`;
